includes/class-default-image.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * Handles Default Image functionality.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Handles Default Image functionality.
+ */
+class Default_Image {
+	/**
+	 * Form field name.
+	 */
+	const FIELD_NAME = 'is_default_image';
+
+	/**
+	 * Option name.
+	 */
+	const OPTION_NAME = 'newspack_default_image_url';
+
+	/**
+	 * Add hooks.
+	 */
+	public static function init() {
+		add_filter( 'attachment_fields_to_save', [ __CLASS__, 'save_attachement_settings' ], 10, 2 );
+		add_filter( 'attachment_fields_to_edit', [ __CLASS__, 'add_attachment_field' ], 10, 2 );
+		add_action( 'template_redirect', [ __CLASS__, 'handle_not_found_image' ] );
+	}
+
+	/**
+	 * Handle not found image.
+	 */
+	public static function handle_not_found_image() {
+		if ( is_404() ) {
```

Comment:
Non-blocking nit: The conditionals here are nested pretty deep. We could return early and have less nesting
`! is_404() || empty( $_SERVER['REQUEST_URI'] )`

Reply:
That's a great suggestions, thanks! Changed in 320e9f82f677